crates/ra_cfg/src/cfg_expr.rs

```diff
@@ -33,6 +33,33 @@ impl CfgExpr {
             CfgExpr::Not(pred) => pred.fold(query).map(|s| !s),
         }
     }
+
+    /// Return minimal features needed
+    pub fn minimal_features_needed(&self) -> Vec<SmolStr> {
+        let mut features = vec![];
+        self.collect_minimal_features_needed(&mut features);
+
+        features
+    }
+
+    fn collect_minimal_features_needed(&self, features: &mut Vec<SmolStr>) {
+        match self {
+            CfgExpr::KeyValue { key, value } if key == "feature" => features.push(value.clone()),
```

Comment:
`cfg` module by design doesn't know about cargo "features". So this code should live elsewhere, at the point where we have information about Cargo. I think `rust-analyzer` crate would be the most approporitate location.